iodata/formats/json.py

```diff
@@ -380,15 +386,41 @@ def _parse_topology_keys(mol: dict, lit: LineIterator) -> dict:
         "id",
         "extras",
     }
-    parsed_keys = molecule_keys.intersection(mol.keys())
-    for key in parsed_keys:
-        del mol[key]
-    if len(mol) > 0:
-        topology_dict["extra"]["unparsed"] = mol
+    passthrough_keys = _find_passthrough_keys(mol, molecule_keys)
+    if passthrough_keys:
+        topology_dict["extra"]["unparsed"] = passthrough_keys
 
     return topology_dict
 
 
+def _find_passthrough_keys(result: dict, keys: set) -> dict:
+    """Find all keys not specified for a given schema.
+
+    Parameters
+    ----------
+    result
+        The JSON dict loaded from file.
+    keys
+        The set of expected keys for a given schema type.
+
+    Returns
+    -------
+    passthrough_dict
+        All unparsed keys remaining in the parsed dict.
+    """
+    # Avoid altering original dict
+    result = result.copy()
+
+    passthrough_dict = dict()
+    parsed_keys = keys.intersection(result.keys())
+    for key in parsed_keys:
+        del result[key]
+    if len(result) > 0:
+        passthrough_dict = result
+
+    return passthrough_dict
+
+
 # pylint: disable=unused-argument
 def _load_qcschema_basis(result: dict, lit: LineIterator) -> dict:
```

Comment:
P.S. You can also prefix unused arguments with an underscore. This is a common convention for identifying unused arguments are variables that need to be present for API consistency. Pylint knows this convention and it is a bit more explicit than adding `# pylint: disable=unused-argument`, because the underscores show which variables are unused.